### PR TITLE
Name, Female, Male, Neutral Reciprocal Relationship Health Checks

### PR DESCRIPTION
**Pull request by MatthewBlanski**
Added Health Checks for Name, Female, Male, Neutral Reciprocal Relationship Lookup values

# Critical Changes

# Changes

# Issues Closed

# New Metadata

## Unpackaged Metadata

# Deleted Metadata

# Testing Notes

### CODE REVIEW

File: src/classes/RelationshipLookupModel.cls [hunk 1]
```diff
@@ -42,10 +42,10 @@ public with sharing class RelationshipLookupModel {
 
     // Constructor to set the values
     public RelationshipLookupModel(String name, Boolean isActive, String femaleValue, String maleValue, String neutralValue) {
-        this.name = name;
+        this.name = (String.isBlank(name) ? name : '""');
         this.isActive = isActive;
-        this.femaleValue = femaleValue;
-        this.maleValue = maleValue;
-        this.neutralValue = neutralValue;
+        this.femaleValue = (String.isBlank(femaleValue) ? femaleValue : '""');
```
Comment (m-kinkade) [inline src/classes/RelationshipLookupModel.cls:47]:
```suggestion
        this.femaleValue = (String.isBlank(femaleValue) ? '""' : femaleValue);
```

File: src/classes/RelationshipLookupModel.cls [hunk 2]
```diff
@@ -42,10 +42,10 @@ public with sharing class RelationshipLookupModel {
...
+        this.maleValue = (String.isBlank(maleValue) ? maleValue : '""');
```
Comment (m-kinkade) [inline src/classes/RelationshipLookupModel.cls:48]:
```suggestion
        this.maleValue = (String.isBlank(maleValue) ? '""' : maleValue);
```

File: src/classes/RelationshipLookupModel.cls [hunk 3]
```diff
@@ -42,10 +42,10 @@ public with sharing class RelationshipLookupModel {
...
+        this.neutralValue = (String.isBlank(neutralValue) ? neutralValue : '""');
```
Comment (m-kinkade) [inline src/classes/RelationshipLookupModel.cls:49]:
```suggestion
        this.neutralValue = (String.isBlank(neutralValue) ? '""' : neutralValue);
```

File: src/classes/RelationshipLookupModel_TEST.cls [hunk 4]
```diff
@@ -50,4 +50,21 @@ private class RelationshipLookupModel_TEST {
...
+    * @description Test method to verify that the constructor set the properties with nulls
+    ***************************************************************************************************************************/
+    @isTest
+    private static void constructorNull() {
+
+        Test.startTest();
+            RelationshipLookupModel relLookupModel = new RelationshipLookupModel(null, true, null, null, null);
+        Test.stopTest();
+
+        System.assertEquals('', relLookupModel.name, 'Name property on the RelationshipLookup model should be set with the value that is passed');
```
Comment (m-kinkade) [inline src/classes/RelationshipLookupModel_TEST.cls:64]:
```suggestion
        System.assertEquals('""', relLookupModel.name, 'Name property on the RelationshipLookup model should be set with the value that is passed');
```

File: src/classes/RelationshipLookupModel_TEST.cls [hunk 5]
```diff
@@ -50,4 +50,21 @@ private class RelationshipLookupModel_TEST {
...
+        System.assertEquals(true, relLookupModel.isActive, 'Active property on the RelationshipLookup model should be set with the value that is passed');
+        System.assertEquals('', relLookupModel.femaleValue, 'Female property on the RelationshipLookup model should be set with the value that is passed');
```
Comment (m-kinkade) [inline src/classes/RelationshipLookupModel_TEST.cls:66]:
```suggestion
        System.assertEquals('""', relLookupModel.femaleValue, 'Female property on the RelationshipLookup model should be set with the value that is passed');
```

File: src/classes/RelationshipLookupModel_TEST.cls [hunk 6]
```diff
@@ -50,4 +50,21 @@ private class RelationshipLookupModel_TEST {
...
+        System.assertEquals('', relLookupModel.maleValue, 'Male property on the RelationshipLookup model should be set with the value that is passed');
```
Comment (m-kinkade) [inline src/classes/RelationshipLookupModel_TEST.cls:67]:
```suggestion
        System.assertEquals('""', relLookupModel.maleValue, 'Male property on the RelationshipLookup model should be set with the value that is passed');
```

File: src/classes/RelationshipLookupModel_TEST.cls [hunk 7]
```diff
@@ -50,4 +50,21 @@ private class RelationshipLookupModel_TEST {
...
+        System.assertEquals('', relLookupModel.neutralValue, 'Neutral property on the RelationshipLookup model should be set with the value that is passed');
```
Comment (m-kinkade) [inline src/classes/RelationshipLookupModel_TEST.cls:68]:
```suggestion
        System.assertEquals('""', relLookupModel.neutralValue, 'Neutral property on the RelationshipLookup model should be set with the value that is passed');
```

File: src/classes/RelationshipLookupModel.cls (hunk 1 above)
Comment (m-kinkade) [reply src/classes/RelationshipLookupModel.cls:47]: The logic in this statement is backwards, I've included a suggestion, but we could also use String.isNotBlank instead.

File: src/classes/RelationshipLookupModel_TEST.cls (hunk 4 above)
Comment (m-kinkade) [reply src/classes/RelationshipLookupModel_TEST.cls:64]: Just need to update the assertions in the test class to check for the empty string surrounded by double quotes.

File: src/classes/RelationshipLookupModel.cls (hunk 1 above)
Comment (MatthewBlanski) [reply src/classes/RelationshipLookupModel.cls:47]: Thank you!
